### PR TITLE
add mask for raster tiles from stream

### PR DESCRIPTION
**Pull request by jdroenner**

### CODE REVIEW

**Comment by jdroenner** [comment]
[Bug: EnMap NoData wird in XArray nicht als NoData angezeigt.](https://trello.com/c/7VAx1RkU/1432-bug-enmap-nodata-wird-in-xarray-nicht-als-nodata-angezeigt)

File: geoengine/workflow.py [hunk 1]
```diff
@@ -566,10 +566,19 @@ def create_xarray(record_batch: pa.RecordBatch) -> xr.DataArray:
...
+            numpy_data = arrow_array.to_numpy(
+                zero_copy_only=False,  # cannot zero-copy as soon as we have nodata values
+            )
+            numpy_mask = arrow_array.is_null(
+                nan_is_null=False  # nan is not no data
+            ).to_numpy(
+                zero_copy_only=False  # cannot zero-copy with bools
+            )
+
+            numpy_masked_data = np.ma.masked_array(numpy_data, mask=numpy_mask).reshape(x_size, y_size)
```
Comment (ChristianBeilschmidt) [inline geoengine/workflow.py:578]: Thanks for finding this out.

Could you add a test that verifies that I don't change it back for optimization reasons in the future?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: geoengine/workflow.py (hunk 1 above)
Comment (jdroenner) [reply geoengine/workflow.py]: well it was a bit more complex but now you can choose how to use the data. For Xarray the data is still converted to float and nan is set as "nodata". For numpy we can now produce masked arrays with the correct type.

**Review by ChristianBeilschmidt — APPROVED** [review]

File: geoengine/raster.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            self.geo_transform.x_max(self.size_x),
+            self.geo_transform.y_max,
+        )
+
+    def spatial_resolution(self) -> gety.SpatialResolution:
+        return self.geo_transform.spatial_resolution()
+
+    @staticmethod
+    def from_ge_record_batch(record_batch: pa.RecordBatch) -> RasterTile2D:
+        '''Create a RasterTile2D from an Arrow record batch recieved from the Geo Engine'''
```
Comment (1lutz) [inline geoengine/raster.py:178]: There is a small typo:
recieved --> received

**Pull request merged**